src/snmalloc/backend_helpers/lockrange.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "../ds/ds.h"
+#include "empty_range.h"
+
+namespace snmalloc
+{
+  /**
+   * Protect the ParentRange with a spin lock.
```

Comment:
It might be worth a comment that this is intended to have a `StaticRange` as a transitive child, otherwise we'll end up with a whole lot of uncontended locks.